File: samples/java/03sqrt/sqrt.java
public class sqrt {
    final static double accuracy = 0.0000001;

    public static double findSqrt(double number) {
        double x = number;
        double y = 1;
        while (x-y > accuracy) {
            x = (x + y) / 2;
            y = number / x;
        }
        return x;
    }
    
    public static void main(String[] args) {
        if (args.length == 0) {
            System.err.println("Error: Usage 'java sqrt <number> [<number> ...]'");
            System.exit(1);
        }
        for (String arg : args) {
            double number = 0;
            try {
                number = Double.parseDouble(arg);
            } catch (java.lang.NumberFormatException e) {
                System.err.println(String.format("Error: Expected a number, received '%s'", arg));
                System.exit(2);
            }
            System.out.println(String.format("%f => %f", number, findSqrt(number)));
        }
    }

}

File: samples/kotlin/03sqrt/sqrt.kt
fun findSqrt(num: Double): Double {
    var x = num;
    var y: Double = 1.0;
    val e: Double = 0.000001;
    while (x-y > e) {
        x = (x + y) / 2;
        y = num / x;
    }
    return x;
}

fun main(args: Array<String>) {
    if (args.size == 0) {
        System.err.println("Error: Usage `kotlin sqrt <number> [<number> ...]`");
        System.exit(1);
    }
    for (arg in args) {
        var number: Double = 0.0;
        try {
            number = arg.toDouble();
        } catch (e: NumberFormatException) {
            System.err.println(String.format("Error: Expected to receive a number. Received '%s'.", arg));
            System.exit(1);
        }
        println(String.format("%f => %f", number, findSqrt(number)));
    }
}
